include/pybind11/attr.h

```diff
@@ -412,14 +418,22 @@ template <> struct process_attribute<arg_v> : process_attribute_default<arg_v> {
         }
         r->args.emplace_back(a.name, a.descr, a.value.inc_ref(), !a.flag_noconvert, a.flag_none);
 
-        if (r->has_kwonly_args) process_kwonly_arg(a, r);
+        if (r->has_kw_only_args) process_kw_only_arg(a, r);
     }
 };
 
 /// Process a keyword-only-arguments-follow pseudo argument
-template <> struct process_attribute<kwonly> : process_attribute_default<kwonly> {
-    static void init(const kwonly &, function_record *r) {
-        r->has_kwonly_args = true;
+template <> struct process_attribute<kw_only> : process_attribute_default<kw_only> {
+    static void init(const kw_only &, function_record *r) {
+        r->has_kw_only_args = true;
+    }
+};
+
+/// Process a positional-only-argument maker
+template <> struct process_attribute<pos_only> : process_attribute_default<pos_only> {
+    static void init(const pos_only &, function_record *r) {
+        r->has_pos_only_args = true;
```

Comment:
It was avoidable, yes! Removed. Not sure how to test this properly at compile time (adding a runtime check would be quite easy).

Reply:
Rather than `any_of<std::is_same<pos_only, Extra>...>::value`, we could write write our own `first_of`, and compare if the index of `pos_only` is before the first `arg` or `arg_v`. I can try something out, but I agree it might not be worth it, indeed.

Besides that, we might want to add a check that there's only 1 `kw_only` and only 1 `pos_only`, maximally? There will be weird effects, otherwise, no?

Reply:
There are several possible ways to improve argument handling, but that probably doesn't have to be in this PR. Ideas:
* `(a, *args, b=None)` is valid in Python but not pybind11.
* `(a, /, **kwargs)` allows "a=" to go into kwargs
* `(a, / a)` isn't valid in Python, but we could allow it
* We could check to make sure `py::pos_only()` is not at the beginning and `py::kw_only()` is not at the end.
* We could have a nice error if either was given twice or in the wrong order.

Reply:
OK, then there are more problems than I thought, yes.

> `(a, /, **kwargs)` allows "a=" to go into kwargs

This works in Python 3.8 as well, though:

```pycon
>>> def f(a, /, **kwargs): pass
... 
>>> f(21)
>>> f(21, a=42)
>>> 
```

Reply:
Everything i listed is valid in Python 3.8 except `(a, /, a)` (after all, what would `a` be?). We could allow it, though, since the name is not the same as the C++ argument for us.

Reply:
Oh, then I misunderstood.

> We could allow it, though, since the name is not the same as the C++ argument for us.

True, but it would be really bad design, and should always easily fixable, I'd say. So sticking with Python-conventions would make things easier, I'd say.